wire-ios-request-strategy/Sources/Payloads/Processing/PayloadProcessing+Conversation.swift

```diff
@@ -189,6 +200,20 @@ extension Payload.Conversation {
         }
     }
 
+    func updateUsersInSystemMessage(with failedUsers: [QualifiedID],
+                                    conversation: ZMConversation,
+                                    type: ZMSystemMessageType,
+                                    context: NSManagedObjectContext) {
+        guard let systemMessage = conversation.firstSystemMessage(for: type, in: context) else {
+            return
+        }
+        for qualifiedID in failedUsers {
+            if let user = ZMUser.fetch(with: qualifiedID.uuid, domain: qualifiedID.domain, in: context) {
+                systemMessage.users.remove(user)
+            }
```

Comment:
You've already fetched the failed users at the call site (actually just after), so I suggest to inject them here already so you don't need to do it again.

Reply:
done